avax-synapse-stablepool/contracts/Strategy.sol

```diff
@@ -0,0 +1,484 @@
+
+        emit Cloned(newStrategy);
+    }
+
+
+    function isWantWETH() internal view returns (bool) {
+        return address(want) == address(weth);
+    }
+
+    function name() external override view returns (string memory) {
+        return strategyName;
+    }
+
+
+
+    function updateMinTimePerInvest(uint256 _minTimePerInvest) public onlyAuthorized {
+        minTimePerInvest = _minTimePerInvest;
+    }
+
+    function updateMaxSingleInvest(uint256 _maxSingleInvest) public onlyAuthorized {
+        maxSingleInvest = _maxSingleInvest;
+    }
+
+    function updateSlippageProtectionIn(uint256 _slippageProtectionIn) public onlyAuthorized {
+        slippageProtectionIn = _slippageProtectionIn;
+    }
+
+    function updateSlippageProtectionOut(uint256 _slippageProtectionOut) public onlyAuthorized {
+        slippageProtectionOut = _slippageProtectionOut;
+    }
+
+    function updateWithdrawProtection(bool _withdrawProtection) public onlyAuthorized {
+        withdrawProtection = _withdrawProtection;
+    }
+
+    function delegatedAssets() public override view returns (uint256) {
+        return vault.strategies(address(this)).totalDebt;
+    }
+
+    function lpTokenToWant(uint256 tokens) public view returns (uint256) {
+        if (tokens == 0) {
+            return 0;
+        }
+
+        //we want to choose lower value of virtual price and amount we really get out
+        //this means we will always underestimate current assets. 
+        uint256 virtualOut = virtualPriceToWant().mul(tokens).div(1e18);
+
+        uint256 realOut = swapPool.calculateRemoveLiquidityOneToken(tokens, curveId);
+
+        return Math.min(virtualOut, realOut);
+    }
+
+    function virtualPriceToWant() public view returns (uint256) {
+        uint256 virtualPrice = basePool.getVirtualPrice();
+
+        if (want_decimals < 18) {
+            return virtualPrice.div(10 ** (uint256(uint8(18) - want_decimals)));
+        } else {
+            return virtualPrice;
+        }
+    }
+    
+    function lpTokensInYVault() public view returns (uint256) {
+        uint256 balance = yvToken.balanceOf(address(this));
+
+        if (yvToken.totalSupply() == 0) {
+            //needed because of revert on priceperfullshare if 0
+            return 0;
+        }
+
+        uint256 pricePerShare = yvToken.pricePerShare();
+        //curve tokens are 1e18 decimals
+        return balance.mul(pricePerShare).div(1e18);
+    }
+
+    function nativeToWant(uint256 _amount) public view override virtual returns (uint256) {
+        if(_amount == 0) {
+            return _amount;
+        }
+        address[] memory path = new address[](2);
+        path[0] = address(weth);
+        path[1] = address(want);
+
+        uint256[] memory amounts = IUniswapV2Router02(router).getAmountsIn(_amount, path);
```

Comment:
This should be getAmountsOut, we are trying to see how much "want" can we get out of exact amount token in